test/networkd-test.py

```diff
@@ -1095,6 +1095,134 @@ def get_tz():
             self.show_journal('systemd-timedated.service')
             self.fail(f'Timezone: {tz}, expected: Pacific/Honolulu')
 
+    def do_test_wait_online_dns(
```

Comment:
Not bad, but nowadays new test cases are added in test/test-network/systemd-networkd-tests.py . Could you move the test case into it?
Otherwise, the test case is only tested on Ubuntu CIs, while the new test runs on many distro CIs.